src/modules/form/hooks/useDynamicFormFields.tsx

```diff
@@ -0,0 +1,100 @@
+import { omit } from 'lodash-es';
+import { useCallback, useMemo, useState } from 'react';
+
+import DynamicFormFields, {
+  isShown,
+  Props as DynamicFormFieldsProps,
+} from '../components/DynamicFormFields';
+import { addDescendants, FormValues } from '../util/formUtil';
+
+import useComputedData from './useComputedData';
+
+import { FormDefinitionJson, FormItem } from '@/types/gqlTypes';
+
+const useDynamicFormFields = ({
+  definition,
+  initialValues,
+  bulk,
+}: {
+  definition: FormDefinitionJson;
+  initialValues?: Record<string, any>;
+  bulk?: boolean;
+}) => {
+  const {
+    items: itemMap,
+    autofillMap: autofillDependencyMap,
+    enabledMap: enabledDependencyMap,
+    disabled: initiallyDisabledLinkIds = [],
+  } = useComputedData({ definition, initialValues });
+
+  const [disabledLinkIds, setDisabledLinkIds] = useState(
+    initiallyDisabledLinkIds
+  );
+  const [values, setValues] = useState<FormValues>(
+    Object.assign({}, initialValues)
+  );
+
+  const getCleanedValues = useCallback(() => {
+    if (!definition) return values;
+    const excluded = addDescendants(disabledLinkIds, definition);
+    return omit(values, excluded);
+  }, [definition, disabledLinkIds, values]);
+
+  const shouldShowItem = useCallback(
+    (item: FormItem) => isShown(item, disabledLinkIds),
+    [disabledLinkIds]
+  );
```

Comment:
`shouldShowItem` feels somewhat unnecessary to pass through this hook since it's only used in one place